### ISSUE TITLE
refresh indexes before download

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dbarra/xdocreport#115]
```
What steps will reproduce the problem?
1. Generate a document, either docx or odt which includes indexes for subtitles 
and for included images.
2. Download via browser and open either with Word or with OpenOffice.
3. The index requires a refresh before being utile and including the subtitles 
and images. This is also true for image numbering.

What is the expected output? What do you see instead?
Being able to refresh the document on the server before streaming to the 
browser. This is specially important if the generated document shall be 
converted to a PDF before streaming to the requester.

What version of the product are you using? On what operating system?
Xdocreport 0.97 on application Server running on Linux and or Win7.

Please provide any additional information below.
What is the hidden difference between the following code snippets, and what is 
the preferred form?
Code snippet:
Options options = Options.getTo(ConverterTypeTo.PDF).via(
                    ConverterTypeVia.ITEXT);
            report.convert(context, options, out);
And from the web site example:
// 1) Load ODT into ODFDOM OdfTextDocument 
InputStream in= new FileInputStream(new File("HelloWord.odt"));
OdfTextDocument document = OdfTextDocument.loadDocument(in);

// 2) Prepare PDFViaIText options
PDFViaITextOptions options = PDFViaITextOptions.create();

// 3) Convert OdfTextDocument to PDF via IText
OutputStream out = new FileOutputStream(new File("HelloWord.pdf"));
ODF2PDFViaITextConverter.getInstance().convert(document, out, options);



```

Original issue reported on code.google.com by `juerg.br...@gmail.com` on 9 May 2012 at 2:48

**Comment by GoogleCodeExporter** [comment]
```
Hi Juerg,

Im' sorry but I don't understand your problem?
Could you please attach some sample. It seems that it's a problem with PDF 
conversion?

Regards Angelo
```

Original comment by `angelo.z...@gmail.com` on 9 May 2012 at 7:12
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Hi Angelo,
It's related to pdf generation too, but only in a second step.
I attached two documents generated with xdocreport with dummy content. The 
important part is the table of content and the table of figures. The first 
shows the table of content as it is received from the download. The second 
after a manual refresh has been done within Word, or OpenOffice in the related 
case.
The aim is to do this refresh on the server before the document is streamed to 
the browser. If it's possible of course!
To generate a pdf with a Table of Content showing nonsense is not very useful. 
So before starting to try to convert either a docx or an odt document to a pdf, 
I must be sure to have the document in a usable form.
If this is not possible, I see only the possibility to disable the pdf 
generation for the moment and to come back on this only much later when 
customers start to cry loud for the feature! :))

P.S. I have MonteRosa running now on Glassfish, WLS, JBoss and Websphere using 
either MySQL, Oracle or DB2! Left work is to look for a new name, as the 
developing label "MonteRosa" is used by other People and I guess, I should take 
the AppEngine as a target too!
```

Original comment by `juerg.br...@gmail.com` on 10 May 2012 at 6:02

**Comment by GoogleCodeExporter** [comment]
```
Hi Juerg,

Ok I understand. The problem is the generation of the table content.
For the moment, I have not idea to manage that.

If you have an idea, don't hesiate to tell me.

Regards Angelo
```

Original comment by `angelo.z...@gmail.com` on 10 May 2012 at 12:50

**Comment by GoogleCodeExporter** [comment]
```
Hi Angelo,
I neither do, but I had to ask you for the case of you having an Idea!
Of course, if I got an Idea of how to achieve it, I'll tell you. But meanwhile 
I postpone this issue as announced. 
- Pour le moment j’ai d’autre chat à fouetter! :))
Regards 
Juerg
```

Original comment by `juerg.br...@gmail.com` on 10 May 2012 at 1:21

**Comment by GoogleCodeExporter** [comment]
```
Issue 126 has been merged into this issue.
```

Original comment by `angelo.z...@gmail.com` on 7 Aug 2014 at 9:01

**Comment by GoogleCodeExporter** [comment]
```
I close this issue because XDocReport can do that. Perhaps with POI?
```

Original comment by `angelo.z...@gmail.com` on 7 Aug 2014 at 9:02
- Changed state: **WontFix**

**Comment by GoogleCodeExporter** [comment]
```
cleanup
```

Original comment by `pascal.leclercq` on 10 Oct 2014 at 7:37
